File: Models/Thread.cs
using System.Collections.Generic;

namespace WebForums.Models
{
	public class Thread
	{
		public int ID { get; set; }
		public virtual Forum Forum { get; set; }
		public string Title { get; set; }
		public User Starter { get; set; }
		public virtual ICollection<Post> Posts { get; set; }
	}
}

File: src/entities/Thread.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany } from "typeorm";
import { Forum } from "./Forum";
import { User } from "./User";
import { Post } from "./Post";

@Entity()
export class Thread {
    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(() => Forum, forum => forum.threads)
    forum: Forum;

    @Column()
    title: string;

    @ManyToOne(() => User, user => user.threads)
    starter: User;

    @OneToMany(() => Post, post => post.thread)
    posts: Post[];
}